pyccel/parser/semantic.py

```diff
@@ -4799,3 +4802,50 @@ def _build_CmathPhase(self, func_call):
         else:
             self.insert_import('math', AsName(MathAtan2, 'atan2'))
             return MathAtan2(PythonImag(var), PythonReal(var))
+
+    def _visit_SetUpdate(self, expr):
+        """
+        Method to navigate the syntactic DottedName node of an `update()` call.
+
+        The purpose of this `_visit` method is to construct new nodes from a syntactic 
+        DottedName node. It checks the type of the iterable passed to `update()`.
+        If the iterable is an instance of `PythonList`, `PythonSet` or `PythonTuple`, it constructs 
+        a CodeBlock node where its body consists of `SetAdd` objects with the 
+        elements of the iterable. If not, it attempts to construct a syntactic `For` 
+        loop to iterate over the iterable object and added its elements to the set 
+        object. Finally, it passes to a `_visit()` call for semantic parsing.
+    
+        Parameters
+        ----------
+        expr : DottedName
+            The syntactic DottedName node that represent the call to `.update()`
+
+        Returns
+        -------
+        PyccelAstNode
+            CodeBlock or For containing SetAdd objects.
+        """
+        iterable = expr.name[1].args[0].value
+        if isinstance(iterable, (PythonList, PythonSet, PythonTuple)):
+            list_variable = self._visit(expr.name[0])
+            added_list = self._visit(iterable)
+            try:
+                store = [SetAdd(list_variable, a) for a in added_list]
+            except TypeError as e:
+                msg = str(e)
+                errors.report(msg, symbol=expr, severity='fatal')
+            return CodeBlock(store)
+        else:
+            pyccel_stage.set_stage('syntactic')
+            for_target = self.scope.get_new_name('index')
```

Comment:
Is there any particular reason to name this temporary variable index? I don't think it always represents an index. E.g.:
```python
a = (1,2,3)
b : set[int] = {}
b.update(a)
```
would be translated into:
```python
a = (1,2,3)
b : set[int] = {}
for index in a:
     b.update(index)
```
But `index` represents an element of `a` not an index.
The `current_name` argument of `get_new_name` is optional so it should only be provided if you have a logical name that you would like to use

Reply:
Yes I think this would make sense here

Reply:
Fixed : b3942c9